### PR TITLE
[HTTPCLIENT-2124] NullPointerException in MinimalHttpClient.doExecute(HttpHost, ClassicHttpRequest, HttpContext)

### PR DESCRIPTION
**Pull request by garydgregory**
`NullPointerException` in `org.apache.hc.client5.http.impl.classic.MinimalHttpClient.doExecute(HttpHost, ClassicHttpRequest, HttpContext)`

### CODE REVIEW

**Review by carterkozak — APPROVED** [review]
Looks good to me, thanks!

**Review by michael-o — COMMENTED** [review]
Where is the NPE and why is there no try block in the second test?

**Comment by carterkozak** [comment]
context is nullable where clientContext is not. The adaptClientContext line produces the clientContext from a nullable context.

**Comment by carterkozak** [comment]
As an aside, we’ve used https://github.com/uber/NullAway at work to avoid NPEs with compile time checks, but it can be painful to retrofit into large existing projects.

**Comment by garydgregory** [comment]
On Sun, Oct 25, 2020 at 1:36 PM Carter Kozak <notifications@github.com>
wrote:

> As an aside, we’ve used https://github.com/uber/NullAway at work to avoid
> NPEs with compile time checks, but it can be painful to retrofit into large
> existing projects.
>

I found this one using SpotBugs which you can run as a Maven or Eclipse
plugin.

Gary

File: httpclient5-testing/src/test/java/org/apache/hc/client5/testing/sync/TestMinimalClientRequestExecution.java [hunk 1]
```diff
@@ -98,4 +100,18 @@ public void testNonCompliantURI() throws Exception {
...
 
+    @Test
+    public void testNonCompliantURIWithoutContext() throws Exception {
+        this.server.registerHandler("*", new SimpleService());
+        this.httpclient = HttpClients.createMinimal();
+        final HttpHost target = start();
+
+        for (int i = 0; i < 10; i++) {
+            final HttpGet request = new HttpGet("/");
+            final CloseableHttpResponse response = this.httpclient.execute(target, request);
```
Comment (michael-o) [inline httpclient5-testing/src/test/java/org/apache/hc/client5/testing/sync/TestMinimalClientRequestExecution.java:111]: Why is there no try with resources block?

Comment (garydgregory) [reply httpclient5-testing/src/test/java/org/apache/hc/client5/testing/sync/TestMinimalClientRequestExecution.java]: Fixed! :-)

**Review by michael-o — APPROVED** [review]

**Pull request merged**